app/src/main/java/com/nextcloud/talk/adapters/messages/IncomingPollMessageViewHolder.kt

```diff
@@ -0,0 +1,291 @@
+            true
+        }
+    }
+
+    private fun setPollPreview(message: ChatMessage) {
+        var pollId: String? = null
+        var pollName: String? = null
+
+        if (message.messageParameters != null && message.messageParameters!!.size > 0) {
+            for (key in message.messageParameters!!.keys) {
+                val individualHashMap: Map<String?, String?> = message.messageParameters!![key]!!
+                if (individualHashMap["type"] == "talk-poll") {
+                    pollId = individualHashMap["id"]
+                    pollName = individualHashMap["name"].toString()
+                }
+            }
+        }
+
+        if (pollId != null && pollName != null) {
+            binding.messagePollTitle.text = pollName
+
+            val roomToken = (payload as? MessagePayload)!!.currentConversation.token!!
+            val isOwnerOrModerator = (payload as? MessagePayload)!!.currentConversation.isParticipantOwnerOrModerator
+
+            binding.bubble.setOnClickListener {
+                val pollVoteDialog = PollMainDialogFragment.newInstance(
+                    message.activeUser!!,
+                    roomToken,
+                    isOwnerOrModerator,
+                    pollId,
+                    pollName
+                )
+                pollVoteDialog.show(
+                    (binding.messagePollIcon.context as MainActivity).supportFragmentManager,
+                    TAG
+                )
+            }
+
+            val credentials = ApiUtils.getCredentials(message.activeUser?.username, message.activeUser?.token)
+            ncApi!!.getPoll(
+                credentials,
+                ApiUtils.getUrlForPoll(
+                    message.activeUser?.baseUrl,
+                    roomToken,
+                    pollId
+                )
+            ).subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(object : Observer<PollOverall> {
+                    override fun onSubscribe(d: Disposable) {
+                        // unused atm
+                    }
+
+                    override fun onNext(pollOverall: PollOverall) {
+                        if (pollOverall.ocs!!.data!!.status == 0) {
+                            binding.messagePollSubtitle.text =
+                                context?.resources?.getString(R.string.message_poll_tap_to_vote)
+                        } else {
+                            binding.messagePollSubtitle.text =
+                                context?.resources?.getString(R.string.message_poll_tap_see_results)
+                        }
+                    }
+
+                    override fun onError(e: Throwable) {
+                        Log.e(TAG, "Error while fetching poll", e)
+                    }
+
+                    override fun onComplete() {
+                        // unused atm
+                    }
+                })
```

Comment:
can you post an example for this?

Reply:
For this particular code it would look like this:

```kotlin
 .subscribe(
                    { pollOverall: PollOverall ->
                        if (pollOverall.ocs!!.data!!.status == 0) {
                            binding.messagePollSubtitle.text =
                                context.resources?.getString(R.string.message_poll_tap_to_vote)
                        } else {
                            binding.messagePollSubtitle.text =
                                context.resources?.getString(R.string.message_poll_tap_see_results)
                        }
                    },
                    { e: Throwable -> Log.e(TAG, "Error while fetching poll", e) }
                )
```

Reply:
thanks. Not needed anymore as we now show static text instead to make api calls.